### PR TITLE
Fix v1 vm support

### PR DESCRIPTION
**Pull request by dennisjbell**
# Improvements

- Better support for vm_types, especially in context of upgrading from v1 environments.

  New feature `v1-vm-types` sets up the manifest to use the same VM types as was used in the v1.x versions of the kit, or as close to possible.  This allows you to use your existing cloud config and tuning.  Where the instance group name changes, it will use the vm type that the instance group was migrated from.

  The exceptions are `tcp-router`, which will use the `router` vm type, and `cc-worker` and `credhub`, which continue to use `minimum` as there was no prevous instance group for these functions.

# Bug Fix

- Supplied missing `params.*_vm_types`, warn instead of error if an unknown instance group is specified, so that custom instance groups can be managed in the same way.

  Note:  For known instance groups, the underscores are automatically converted to hyphens to determine the matching instance_group for the specified `*_vm_type` params.  For user specified, you must specify the hyphen or underscore as used in the instance_group.

  Known instance groups are:
    adapter, api, cc-worker, credhub, database, diego-api, diego-cell, doppler, errand, haproxy, log-api, nats, rotate-cc-database-key, router, scheduler, singleton-blobstore, smoke-tests, tcp-router, and uaa

### CODE REVIEW

**Review by rkoster — CHANGES_REQUESTED** [review]
Looks good, but is hard to judge if it actually works without a spec test.
Maybe add a deployment where migrated_v1_env is set in exodus.

**Comment by rkoster** [comment]
@dennisjbell please review and merge https://github.com/dennisjbell/cf-genesis-kit/pull/1 into your feature branch.

File: spec/results/override-vm-types-and-counts-old-names.yml [hunk 1]
```diff
@@ -0,0 +1,2075 @@
...
+        tls:
+          ca: <!{credhub}:credhub_tls.ca!>
+          certificate: <!{credhub}:credhub_tls.certificate!>
+          private_key: <!{credhub}:credhub_tls.private_key!>
+    release: credhub
+  name: credhub
+  networks:
+  - name: cf-core
+  stemcell: default
+  vm_type: minimal
```
[the remark this answers is not in the document]
Comment (dennisjbell) [reply spec/results/override-vm-types-and-counts-old-names.yml:1919]: not with old-names test (there was no credhub instance in the old kit)

**Review by rkoster — APPROVED** [review]

**Comment by rkoster** [comment]
LGTM!

**Review by krutten — APPROVED** [review]
Reviewed changes and test cases.  LGTM

File: MANUAL.md [hunk 2]
```diff
@@ -449,18 +449,71 @@ following:
...
+as was used in the v1.x versions of the kit, or as close to possible.  This
+allows you to use your existing cloud config and tuning.  Where the instance
+group name changes, it will use the vm type that the instance group was
+migrated from.
+
+The exceptions are `tcp-router`, which will use the `router` vm type, and
+`cc-worker` and `credhub`, which continue to use `minimum` as there was no
+prevous instance group for these functions.
+
+If you want to use the de facto vm types after migration, you can specify the
```
Comment (krutten) [inline MANUAL.md:492]: Clarifier:  Should state that `no-v1-vm-types` is a feature so it's clear where it goes.

**Pull request merged**